vms/evm/subscriber.go

```diff
@@ -117,6 +125,28 @@ func (s *subscriber) processBlockRange(
 	return nil
 }
 
+func (s *subscriber) getHeaderByNumberRetryable(headerNumber *big.Int) (*types.Header, error) {
+	var err error
+	var header *types.Header
+	for i := 0; i < rpcMaxRetries; i++ {
+		header, err = s.rpcClient.HeaderByNumber(context.Background(), headerNumber)
+		if err == nil {
+			return header, nil
+		}
+		s.logger.Warn(
+			"Failed to get header by number",
+			zap.String("blockchainID", s.blockchainID.String()),
+			zap.Int("attempt", i),
+			zap.Error(err),
+		)
+		// Sleep if this wasn't the last retry
+		if i < rpcMaxRetries-1 {
+			time.Sleep(time.Duration(i+1) * time.Second)
```

Comment:
nit: it might be more clear to specify the total timeout/sleep duration (10 seconds across 5 retries) as a constant. Any reason why the subscriber timeout is a constant on each retry, but the rpc timeout grows linearly with each retry?

Reply:
I agree but in which direction? constant? Ideally with network resources I like exponential backoffs with jitter but seemed like an overkill here.

Reply:
Exponential backoff has come up in previous discussions, but we've come to similar conclusions that it's overkill. Since the overall number of RPC calls is low, optimizing how they're spaced hasn't been a top priority.

Reply:
I thought we were tracking that already, but apparently not. I created https://github.com/ava-labs/awm-relayer/issues/453 to do so.